include/eve/arch/riscv/tags.hpp

```diff
@@ -0,0 +1,118 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/cardinals.hpp>
+#include <eve/arch/cpu/tags.hpp>
+#include <eve/arch/riscv/predef.hpp>
+#include <eve/concept/scalar.hpp>
+#include <eve/detail/meta.hpp>
+
+#include <climits>
+
+namespace eve
+{
+//================================================================================================
+// ABI tags for all RISCV bits SIMD registers
+//================================================================================================
+template<std::size_t Size> struct rvv_abi_
+{
+  static_assert(CHAR_BIT == 8, "[eve riscv] - For riscv we expect CHAR_BIT to be 8");
+  static constexpr std::size_t MaxLmul         = 8;
+  static constexpr std::size_t bits            = Size * MaxLmul;
+  static constexpr std::size_t bytes           = bits / 8;
+  static constexpr bool        is_wide_logical = false;
+
+  template<typename Type>
+  static constexpr bool is_full = ((Type::size() * sizeof(typename Type::value_type)) >= bytes);
+
+  template<typename Type>
+  static constexpr std::size_t fundamental_cardinal = Size / 8 / sizeof(Type);
+
+  template<typename Type>
+  static constexpr std::size_t expected_cardinal = Size / 8 / sizeof(Type) * MaxLmul;
+
+  template<typename Type> static consteval int get_min_frac_lmul()
+  {
+    auto type_size = sizeof(Type);
+    if( type_size == 1 ) return 8;
+    if( type_size == 2 ) return 4;
+    if( type_size == 4 ) return 2;
+    // for bigger types we can not use frac lmul.
+    // Return bigger than size of vector register value.
+    return Size + 1;
+  }
+
+  template<typename Type> static consteval int get_max_frac_lmul()
+  {
+    auto type_size = sizeof(Type);
+    if( type_size == 1 ) return 2;
+    if( type_size == 2 ) return 2;
+    if( type_size == 4 ) return 2;
+    // for bigger types we can not use frac lmul.
+    // Return bigger than size of vector register value.
+    return Size + 1;
+  }
+
+  // return natural lmul if > 0, frac otherwise
+  template<typename Type> static consteval int getLMUL(size_t cardinal)
+  {
+    auto binary_size   = cardinal * sizeof(Type) * 8;
+    auto min_frac_lmul = get_min_frac_lmul<Type>();
+    auto max_frac_lmul = get_max_frac_lmul<Type>();
+
+    auto frac_lmul_border = Size / max_frac_lmul;
+    if( binary_size <= frac_lmul_border )
+    {
+      auto frac_lmul = Size / binary_size;
+      frac_lmul      = std::bit_floor(frac_lmul);
```

Comment:
This requires `<bit>` as per the error in the checks.

Reply:
My bad, added.